### PR TITLE
NEXUS-5049: Fix

### PR DESCRIPTION
**Pull request by cstamas**
See issue for details.

CI:
https://builds.sonatype.org/view/nexus/job/nexus-oss-its-feature/336/
(will fail without last commit)

Scheduled (hence, job No might not be the correct one):
https://builds.sonatype.org/view/nexus/job/nexus-oss-its-feature/337/

### CODE REVIEW

File: nexus/nexus-app/src/main/java/org/sonatype/nexus/configuration/application/DefaultNexusConfiguration.java [hunk 1]
```diff
@@ -542,6 +549,137 @@ public boolean isAnonymousAccessEnabled()
...
 
+    public void setAnonymousAccess( final boolean enabled, final String username, final String password )
+        throws InvalidConfigurationException
+    {
+        if ( enabled )
+        {
+            if ( StringUtils.isBlank( username ) || StringUtils.isBlank( password ) )
+            {
+                throw new InvalidConfigurationException(
+                    "Anonymous access is getting enabled without valid username and/or password!" );
```
Comment (peterlynch) [inline nexus/nexus-app/src/main/java/org/sonatype/nexus/configuration/application/DefaultNexusConfiguration.java:560]: Does this bubble up to the UI? If so should we include our suggested remedy?

File: nexus/nexus-app/src/main/java/org/sonatype/nexus/configuration/application/DefaultNexusConfiguration.java [hunk 2]
```diff
@@ -542,6 +549,137 @@ public boolean isAnonymousAccessEnabled()
...
+            // we did not perform any _change_ against security sofar (we just did reading from it),
+            // so it is okay to bail out at this point
+            getLogger().warn(
+                "XML Realm reported invalid configuration while trying to disable Anonymous user (as part of disabling anonymous access)!",
+                e );
+            throw e;
+        }
+    }
+
+    @Deprecated
```
Comment (peterlynch) [inline nexus/nexus-app/src/main/java/org/sonatype/nexus/configuration/application/DefaultNexusConfiguration.java:682]: Do we add reason?

Comment (cstamas) [inline nexus/nexus-app/src/main/java/org/sonatype/nexus/configuration/application/DefaultNexusConfiguration.java:682]: This is the implementation of an interface. See interface for full description and what to use instead.

File: nexus/nexus-app/src/main/java/org/sonatype/nexus/configuration/application/DefaultNexusConfiguration.java (hunk 1 above)
Comment (cstamas) [inline nexus/nexus-app/src/main/java/org/sonatype/nexus/configuration/application/DefaultNexusConfiguration.java:560]: Yes, it ends up as HTTP 400 Bad Request response.

**Comment by peterlynch** [comment]
other than mentioned,  +1
nice comments in code - useful for later refactoring

**Comment by cstamas** [comment]
CI voted +1 (both slaves):
https://builds.sonatype.org/view/nexus/job/nexus-oss-its-feature/337/

**Comment by nabcos** [comment]
+1

**Comment by adreghiciu** [comment]
+1

**Pull request merged**